packages/client/src/components/App.tsx

```diff
@@ -1,7 +1,73 @@
 import '../styles/reset.css';
+import '../styles/tictactoe.css'
+import React from 'react';
+import { Difficulty, GameStatus, Marker, Player } from '@mapistry/take-home-challenge-shared';
 
-export const App = () => (
-  <main>
-    <div>FIXME: add your solution here!</div>
-  </main>
-);
+import { begin, move } from '../api'
+import { Square } from './Square'
+
+export class App extends React.Component<unknown, GameStatus> {
+  constructor(props: unknown) {
+    super(props)
+    this.state = {
+      board: [],
+      winner: null
+    };
+  }
+
+  componentDidMount() {
+    begin(Difficulty.hard, Player.human).then(res => this.setState(res))
+  }
+
+  handleClickOnSquare(index: number) {
+    const { board } = this.state
+    const updatedBoard = board.slice()
+    updatedBoard[index] = Marker.x
+
+    move(updatedBoard, Difficulty.hard).then(res => this.setState(res))
+  }
+
+  renderSquare(index: number) {
+    const { board, winner } = this.state
+    const shouldDisableButton = winner !== null || board[index] !== null
+    return (<Square isDisabled={shouldDisableButton} handleClick={() => this.handleClickOnSquare(index)} value={board[index]} />)
+  }
+
+  render() {
+    const { winner } = this.state;
+    const messageMap = {
```

Comment:
I don't like that these keys are "magic". I tried a couple of approaches to reuse the `Marker` and `Winner` classes to define the keys here, but was having difficulty. Also, if we were to continue this feature to include the first player setting, then we'd want a different solution here since the roles would change.